cmd/cosign/cli/options/predicate.go

```diff
@@ -42,6 +43,7 @@ var PredicateTypeMap = map[string]string{
 	PredicateSPDX:   in_toto.PredicateSPDX,
 	PredicateLink:   in_toto.PredicateLinkV1,
 	PredicateVuln:   attestation.CosignVulnProvenanceV01,
+	PredicateInToto: in_toto.PayloadType,
```

Comment:
I don't think this is correct. A payload is not the same thing as a predicate. IMO, the "sanity checks" in `verify_attestation.go` are wrong and will never pass. See the first commit in https://github.com/sigstore/cosign/pull/1672.

Reply:
Yes I think you are right, there seems to be an underlying semantic discrepancy here (like in #1249 and #1248), this was a quick first pass at getting the functionality I needed. The code here doesn't seem to be used outside of the verify-attestation code path, so it seemed safe to modify even if it is patchy.

Reply:
we could check if an attestation is in DSSE format (using smth like [this](https://github.com/sigstore/cosign/blob/ba50ee05c87701f532c63ec6d0fd0d7f3c03d34d/cmd/cosign/cli/verify/verify_blob.go#L418)) & then go through a different flow for verification since it doesn't clearly map to the current predicate checks?

Reply:
@priyawadhwa that makes sense I can rework this PR if this is the way to go
@lcarva I tried running with the changes from #1672 and sadly that doesn't help for CUE case as well, if I run with a CUE policy that should deem the attestation invalid (i.e. change from `true` to `false` above) the verification still passes.